string log_msg
string file
---
string response